frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -498,63 +497,89 @@ const defaultFilters = {
+  const [prevProps, setPrevProps] = React.useState(props);
+
+  const [categories, setCategories] = React.useState(
+    categorizeItems(items, itemsSorter, getAvailableCategories(items)),
+  );
+  const [selectedCategoryId, setSelectedCategoryId] = React.useState('all');
+  const [activeFilters, setActiveFilters] = React.useState(defaultFilters);
+  const [filterCounts, setFilterCounts] = React.useState(null);
+  const [filterGroupsShowAll, setFilterGroupsShowAll] = React.useState({});
+  const [groupBy, setGroupBy] = React.useState(groupByTypes ? groupByTypes.None : '');
 
-    this.onUpdateFilters = this.onUpdateFilters.bind(this);
-    this.onFilterChange = this.onFilterChange.bind(this);
-    this.renderFilterGroup = this.renderFilterGroup.bind(this);
-    this.onShowAllToggle = this.onShowAllToggle.bind(this);
-    this.onGroupChange = this.onGroupChange.bind(this);
-  }
+  const getUpdatedState = (selectedCategories, categoryId, filters) => {
+    if (!items) {
+      return;
+    }
+
+    const newCategories = recategorizeItems(
+      items,
+      itemsSorter,
+      filters,
+      keywordCompare,
+      selectedCategories,
+    );
+
+    return {
+      activeFilters: filters,
+      selectedCategoryId: categoryId,
+      categories: newCategories,
+      filterCounts: getFilterGroupCounts(
+        items,
+        itemsSorter,
+        filterGroups,
+        categoryId,
+        filters,
+        newCategories,
+        keywordCompare,
+      ),
+    };
+  };
 
-  componentDidMount() {
-    const { items, filterGroups, getAvailableFilters, groupByTypes } = this.props;
-    const { categories } = this.state;
+  React.useEffect(() => {
     const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
     const activeValues = getActiveValuesFromURL(availableFilters, filterGroups, groupByTypes);
 
-    this.setState({
-      ...this.getUpdatedState(
-        categories,
-        activeValues.selectedCategoryId,
-        activeValues.activeFilters,
-      ),
-      groupBy: activeValues.groupBy,
-    });
-    this.filterByKeywordInput.focus({ preventScroll: true });
-  }
+    const updatedState = getUpdatedState(
+      categories,
+      activeValues.selectedCategoryId,
+      activeValues.activeFilters,
+    );
 
-  componentWillUnmount() {
-    this.unmounted = true;
-  }
+    setCategories(updatedState.categories);
+    setSelectedCategoryId(updatedState.selectedCategoryId);
+    setActiveFilters(updatedState.activeFilters);
+    setFilterCounts(updatedState.filterCounts);
+    setGroupBy(activeValues?.groupBy);
 
-  componentDidUpdate(prevProps) {
-    const { activeFilters, selectedCategoryId, groupBy } = this.state;
-    const {
-      items,
-      itemsSorter,
-      filterGroups,
-      getAvailableCategories,
-      getAvailableFilters,
-    } = this.props;
+    filterByKeywordInput.current.focus({ preventScroll: true });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Suppressing `react-hooks/exhaustive-deps` rule should be avoided unless necessary.

Is the empty deps array here because we want to run the effect once (after initial render), regardless of any potential variables?

What are the deps suggested by the `react-hooks/exhaustive-deps` rule here?

Reply:
> This hook replaces the componentDidMount() function in the class component, so we want it to run once after initial render.

Thanks, I see now why it's done that way.

I'd suggest to extract the whole body of that `React.useEffect` call into a separate local function called `initState` to make this more clear.

```ts
// eslint-disable-next-line react-hooks/exhaustive-deps -- should run once on component mount
React.useEffect(initState, []);
```

Since we want to run this logic just once on component mount, I don't think we need to refactor it any further - we'd either have to ensure that all deps are referentially stable across the whole component lifetime (which is not the case as you mentioned above), or just use `[]` to explicitly indicate our intent. So in this particular case, the `react-hooks/exhaustive-deps` rule override seems to be valid.